plugins/pur/core/src/main/java/org/pentaho/di/repository/pur/PurRepository.java

```diff
@@ -546,8 +577,17 @@ public ObjectId renameRepositoryDirectory( final ObjectId dirId, final Repositor
     String finalParentPath = null;
     String interimFolderPath = null;
     try {
-      RepositoryFile homeFolder = pur.getFile( ClientRepositoryPaths.getUserHomeFolderPath( user.getLogin() ) );
-      RepositoryFile folder = pur.getFileById( dirId.getId() );
+
+      readWriteLock.readLock().lock();
+      RepositoryFile homeFolder;
+      RepositoryFile folder;
+      try {
+        homeFolder = pur.getFile( ClientRepositoryPaths.getUserHomeFolderPath( user.getLogin() ) );
+        folder = pur.getFileById( dirId.getId() );
+      } finally {
+        readWriteLock.readLock().unlock();
```

Comment:
The lock should not be release between this and lower in the same method. The lock should be kept. Make this a write lock.